lib/Frontend/CompilerInvocation.cpp

```diff
@@ -126,6 +130,50 @@ generateOptimizationRemarkRegex(DiagnosticEngine &Diags, ArgList &Args,
   return Pattern;
 }
 
+// Lifted from the clang driver.
+static void PrintArg(raw_ostream &OS, const char *Arg, StringRef TempDir) {
```

Comment:
We really ought to coalesce the other implementations of this…

- `escapeAndPrintString` in Job.cpp
- `escape` in `emitMakeDependenciesIfNeeded` in FrontendTool.cpp